### PR TITLE
onChange should be on this component

### PR DESCRIPTION
**Pull request by stevemao**

### CODE REVIEW

**Comment by stevemao** [comment]
The value of my text input should be controlled by this component itself, instead of the `props` from parent component. Otherwise if the parent forgot the pass value, there is no value when changing the input text. Also parent shouldn't worry about unnecessary `state` update so that `render` function is called when input text changes.

**Comment by stevemao** [comment]
The current example is probably a bad one. Whenever the input text is changed, it renders four time (four instances of AutosizeInput) but only one is needed.

File: src/AutosizeInput.js [hunk 1]
```diff
@@ -58,12 +59,18 @@ const AutosizeInput = React.createClass({
 			placeholderNode.style.letterSpacing = inputStyle.letterSpacing;
 		}
 	},
+	onInputChange (event) {
```
Comment (cameronbourke) [inline src/AutosizeInput.js]: Hey, I noticed you aren't calling `this.props.onChange` that has been passed on props. How would a parent component have access to the value when it changes. I think you need to add the following:

```
this.props.onChange ? this.props.onChange(value);
```

**Comment by cameronbourke** [comment]
Hey @stevemao I've added a note in the files changed.

**Comment by stevemao** [comment]
Thanks @cameronbourke . First of all, this is meant to be alternative of #50 so we don't need to discuss both. And I really wanna go with this one.

I haven't thought too much on `this.props.onChange`. I think instead we could change `this.props.value` so `onChange` will be called automatically.

**Comment by stevemao** [comment]
Sorry, `this.props.value` is changed... Let me do some more testing when I get home.

**Comment by cameronbourke** [comment]
I'm probably just being a little thick. What do you mean by? Thanks!
> I think instead we could change this.props.value so onChange will be called automatically

**Comment by stevemao** [comment]
@cameronbourke I have fixed it.

**Comment by stevemao** [comment]
I was thinking `onChange` would propagate... But seems like it doesn't.

**Comment by cameronbourke** [comment]
@stevemao The main issue with your changes is that the consumer of `<AutosizeInput />` does not have control over the input. If the consumer wanted to stop letting the user type after 10 characters for example, there is no way to do so because `<AutosizeInput />` controls the input and the value in the input would be updated.

**Comment by stevemao** [comment]
@cameronbourke fixed. use `setValue`

**Comment by stevemao** [comment]
ping @cameronbourke @JedWatson

**Comment by JedWatson** [comment]
The whole point of this component is for it to be "controlled" - i.e. to manage state in a higher level component. Like `<input>` components in React itself. I don't think needing to use a ref and call a method is simpler, and it's really not "react-like"

**Comment by stevemao** [comment]
@JedWatson Maybe we could support both controlled/uncontrolled component? Similar to how `<input>` works?

**Comment by JedWatson** [comment]
Going to close this, I'd prefer to leave this as a controlled component long term. Thanks for the input @stevemao

**Pull request closed without merging**